<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Alexander Fisher">

<title>Multiple regression</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="multipleRegression_sta613_files/libs/clipboard/clipboard.min.js"></script>
<script src="multipleRegression_sta613_files/libs/quarto-html/quarto.js" type="module"></script>
<script src="multipleRegression_sta613_files/libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="multipleRegression_sta613_files/libs/quarto-html/popper.min.js"></script>
<script src="multipleRegression_sta613_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="multipleRegression_sta613_files/libs/quarto-html/anchor.min.js"></script>
<link href="multipleRegression_sta613_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="multipleRegression_sta613_files/libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="multipleRegression_sta613_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="multipleRegression_sta613_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="multipleRegression_sta613_files/libs/bootstrap/bootstrap-8303d6ff729cf94d8c0e8f588a4e23e9.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="quarto-light">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#libraries" id="toc-libraries" class="nav-link active" data-scroll-target="#libraries">Libraries</a></li>
  <li><a href="#data-metabolic-assays" id="toc-data-metabolic-assays" class="nav-link" data-scroll-target="#data-metabolic-assays">Data: metabolic assays</a>
  <ul class="collapse">
  <li><a href="#load-data" id="toc-load-data" class="nav-link" data-scroll-target="#load-data">Load data</a></li>
  <li><a href="#exploratory-data-analysis" id="toc-exploratory-data-analysis" class="nav-link" data-scroll-target="#exploratory-data-analysis">Exploratory data analysis</a></li>
  </ul></li>
  <li><a href="#linear-modeling" id="toc-linear-modeling" class="nav-link" data-scroll-target="#linear-modeling">Linear modeling</a>
  <ul class="collapse">
  <li><a href="#the-simple-regression-model-and-notation" id="toc-the-simple-regression-model-and-notation" class="nav-link" data-scroll-target="#the-simple-regression-model-and-notation">The simple regression model and notation</a></li>
  <li><a href="#writing-simple-regression-in-matrix-form" id="toc-writing-simple-regression-in-matrix-form" class="nav-link" data-scroll-target="#writing-simple-regression-in-matrix-form">Writing simple regression in matrix form</a></li>
  <li><a href="#key-assumptions" id="toc-key-assumptions" class="nav-link" data-scroll-target="#key-assumptions">Key assumptions</a></li>
  <li><a href="#picture-of-simple-vs-multiple-regression-offline" id="toc-picture-of-simple-vs-multiple-regression-offline" class="nav-link" data-scroll-target="#picture-of-simple-vs-multiple-regression-offline">Picture of simple vs multiple regression (offline)</a></li>
  </ul></li>
  <li><a href="#linear-modeling-in-r" id="toc-linear-modeling-in-r" class="nav-link" data-scroll-target="#linear-modeling-in-r">Linear modeling in R</a></li>
  <li><a href="#fitted-model-output" id="toc-fitted-model-output" class="nav-link" data-scroll-target="#fitted-model-output">Fitted model output</a></li>
  <li><a href="#return-to-metabolic-assay-data" id="toc-return-to-metabolic-assay-data" class="nav-link" data-scroll-target="#return-to-metabolic-assay-data">Return to Metabolic Assay Data</a>
  <ul class="collapse">
  <li><a href="#linear-model-without-batch-pair-effects" id="toc-linear-model-without-batch-pair-effects" class="nav-link" data-scroll-target="#linear-model-without-batch-pair-effects">Linear model without batch (pair) effects</a></li>
  <li><a href="#linear-model-with-batch-pair-effects" id="toc-linear-model-with-batch-pair-effects" class="nav-link" data-scroll-target="#linear-model-with-batch-pair-effects">Linear model with batch (pair) effects</a></li>
  </ul></li>
  <li><a href="#testing" id="toc-testing" class="nav-link" data-scroll-target="#testing">Testing</a>
  <ul class="collapse">
  <li><a href="#key-quantities" id="toc-key-quantities" class="nav-link" data-scroll-target="#key-quantities">Key quantities:</a></li>
  <li><a href="#f-test-for-the-comparison-of-nested-models" id="toc-f-test-for-the-comparison-of-nested-models" class="nav-link" data-scroll-target="#f-test-for-the-comparison-of-nested-models">F-test for the comparison of nested models</a></li>
  <li><a href="#example-f-test-for-the-genotype-effect" id="toc-example-f-test-for-the-genotype-effect" class="nav-link" data-scroll-target="#example-f-test-for-the-genotype-effect">Example: F-test for the genotype effect</a></li>
  </ul></li>
  <li><a href="#dose-as-continuous-covariate" id="toc-dose-as-continuous-covariate" class="nav-link" data-scroll-target="#dose-as-continuous-covariate">Dose as continuous covariate</a>
  <ul class="collapse">
  <li><a href="#curvilinear-dose" id="toc-curvilinear-dose" class="nav-link" data-scroll-target="#curvilinear-dose">Curvilinear dose</a></li>
  </ul></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Multiple regression</h1>
<p class="subtitle lead">STA613</p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Alexander Fisher </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Follow along
</div>
</div>
<div class="callout-body-container callout-body">
<p>Access data and scripts here: <a href="https://duke.box.com/s/6c56uvvngncd6pr080xoelm6sfdgjimu">https://duke.is/sta613-fisher</a></p>
</div>
</div>
<section id="libraries" class="level2">
<h2 class="anchored" data-anchor-id="libraries">Libraries</h2>
<p>In these notes, we use the following libraries:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mgcv)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="data-metabolic-assays" class="level2">
<h2 class="anchored" data-anchor-id="data-metabolic-assays">Data: metabolic assays</h2>
<p>Data come from <span class="citation" data-cites="fisher2018mitochondrial">Fisher-Wellman et al. (<a href="#ref-fisher2018mitochondrial" role="doc-biblioref">2018</a>)</span>. Major goal: bridge the gap between molecular signatures and functional mitochondrial bioenergetics by directly measuring mitochondrial energy flux.</p>
<ul>
<li>Skeletal muscle mitochondria are isolated from either the NT or Tg mice and the same pool of mitochondrial suspension is used for JO2, dPsi, Redox, and e-1 Leak measures.</li>
<li>Each of the 6 substrates (GM, PM, PMPc, PMOc, PcM and OcM) assesses the performance of various metabolic pathways.</li>
<li>You can think of each assay as a type of “dose-response” curve where varying the <span class="math inline">\(\Delta\)</span>G<span class="math inline">\(_{\mathrm{ATP}}\)</span> (-12.95 through 14.49) is like a dose.</li>
<li><code>JO2</code> measures Flux of Oxygen; indicator of how much “work” the mitochondria are capable of performing under a given fuel condition and energy charge.</li>
</ul>
<section id="load-data" class="level3">
<h3 class="anchored" data-anchor-id="load-data">Load data</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>o2 <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../data_s/o2.csv"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>o2<span class="sc">$</span>dose <span class="ot">&lt;-</span> <span class="fu">factor</span>(o2<span class="sc">$</span>dose, <span class="at">levels =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">13.65</span>, <span class="sc">-</span><span class="fl">13.95</span>, <span class="sc">-</span><span class="fl">14.19</span>, <span class="sc">-</span><span class="fl">14.36</span>, <span class="sc">-</span><span class="fl">14.49</span>))</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>o2<span class="sc">$</span>pair <span class="ot">&lt;-</span> <span class="fu">as.character</span>(o2<span class="sc">$</span>pair)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="exploratory-data-analysis" class="level3">
<h3 class="anchored" data-anchor-id="exploratory-data-analysis">Exploratory data analysis</h3>
<p>Take a look at the data:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(o2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 340
Columns: 7
$ y         &lt;dbl&gt; 11946.57, 13218.91, 12843.75, 13296.75, 11647.96, 11642.01, …
$ substrate &lt;chr&gt; "GM", "GM", "GM", "GM", "GM", "GM", "GM", "GM", "GM", "GM", …
$ subject   &lt;chr&gt; "NT1", "NT2", "NT3", "NT4", "NT5", "NT6", "Tg1", "Tg2", "Tg3…
$ dose      &lt;fct&gt; -13.65, -13.65, -13.65, -13.65, -13.65, -13.65, -13.65, -13.…
$ pair      &lt;chr&gt; "1", "2", "3", "4", "5", "6", "1", "2", "3", "4", "5", "6", …
$ geno      &lt;chr&gt; "NT", "NT", "NT", "NT", "NT", "NT", "Tg", "Tg", "Tg", "Tg", …
$ isTg      &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 1, …</code></pre>
</div>
</div>
<p>Marginal summaries:</p>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true" href="">Substrate</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false" href="">Subject</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-3" role="tab" aria-controls="tabset-1-3" aria-selected="false" href="">Dose</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-4" role="tab" aria-controls="tabset-1-4" aria-selected="false" href="">Genotype</a></li></ul>
<div class="tab-content">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(o2<span class="sc">$</span>substrate, <span class="at">useNA =</span> <span class="st">"always"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
  GM  OcM  PcM   PM PMOc PMPc &lt;NA&gt; 
  60   50   50   60   60   60    0 </code></pre>
</div>
</div>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(o2<span class="sc">$</span>subject, <span class="at">useNA =</span> <span class="st">"always"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
 NT1  NT2  NT3  NT4  NT5  NT6  Tg1  Tg2  Tg3  Tg4  Tg5  Tg6 &lt;NA&gt; 
  30   30   30   30   30   20   30   30   30   30   30   20    0 </code></pre>
</div>
</div>
</div>
<div id="tabset-1-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-3-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(o2<span class="sc">$</span>dose, <span class="at">useNA =</span> <span class="st">"always"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
-13.65 -13.95 -14.19 -14.36 -14.49   &lt;NA&gt; 
    68     68     68     68     68      0 </code></pre>
</div>
</div>
</div>
<div id="tabset-1-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-4-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(o2<span class="sc">$</span>geno, o2<span class="sc">$</span>pair, <span class="at">useNA =</span> <span class="st">"always"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>      
        1  2  3  4  5  6 &lt;NA&gt;
  NT   30 30 30 30 30 20    0
  Tg   30 30 30 30 30 20    0
  &lt;NA&gt;  0  0  0  0  0  0    0</code></pre>
</div>
</div>
</div>
</div>
</div>
<section id="exploring-response-variable" class="level4">
<h4 class="anchored" data-anchor-id="exploring-response-variable">Exploring response variable</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(o2<span class="sc">$</span>y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   1197    4412    5591    6509    7721   15284 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="at">las =</span> <span class="dv">1</span>)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(o2<span class="sc">$</span>y, <span class="at">nclass =</span> <span class="dv">20</span>)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fu">qqnorm</span>(o2<span class="sc">$</span>y)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">a =</span> <span class="fu">mean</span>(o2<span class="sc">$</span>y),</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">b =</span> <span class="fu">sd</span>(o2<span class="sc">$</span>y),</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">col =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="multipleRegression_sta613_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="linear-modeling" class="level2">
<h2 class="anchored" data-anchor-id="linear-modeling">Linear modeling</h2>
<section id="the-simple-regression-model-and-notation" class="level3">
<h3 class="anchored" data-anchor-id="the-simple-regression-model-and-notation">The simple regression model and notation</h3>
<p><span class="math display">\[
y = \beta_0 + \beta_1 x + \epsilon
\]</span></p>
<ul>
<li><p><span class="math inline">\(y\)</span>: the <strong>outcome</strong> variable. Also called the “response” or “dependent variable”. In prediction problems, this is what we are interested in predicting.</p></li>
<li><p><span class="math inline">\(x\)</span>: the <strong>predictor</strong>. Also commonly referred to as “regressor”, “independent variable”, “covariate”, “feature”, “the data”.</p></li>
<li><p><span class="math inline">\(\beta_0\)</span>, <span class="math inline">\(\beta_1\)</span> are called “constants” or <strong>coefficients</strong>. They are fixed numbers. These are <strong>population parameters</strong>. <span class="math inline">\(\beta_0\)</span> has another special name, “the intercept”.</p></li>
<li><p><span class="math inline">\(\epsilon\)</span>: the <strong>error</strong>. This quantity represents observational error, i.e.&nbsp;the difference between our observation and the true population-level expected value: <span class="math inline">\(\beta_0 + \beta_1 x\)</span>.</p></li>
</ul>
<p>Effectively this model says our data <span class="math inline">\(y\)</span> is linearly related to <span class="math inline">\(x\)</span> but is not perfectly observed due to some error.</p>
</section>
<section id="writing-simple-regression-in-matrix-form" class="level3">
<h3 class="anchored" data-anchor-id="writing-simple-regression-in-matrix-form">Writing simple regression in matrix form</h3>
<p><span class="math display">\[
\boldsymbol{y}= \boldsymbol{X}\beta + \boldsymbol{\varepsilon},
\]</span></p>
<p>where</p>
<ul>
<li><span class="math inline">\(\boldsymbol{y}\in \mathbb{R}^n\)</span></li>
<li><span class="math inline">\(\boldsymbol{X}\in \mathbb{R}^{n \times p}\)</span></li>
<li><span class="math inline">\(\beta \in \mathbb{R}^p\)</span></li>
<li><span class="math inline">\(\boldsymbol{\varepsilon}\in \mathbb{R}^n\)</span></li>
</ul>
<p>and <span class="math inline">\(p = 2\)</span>, i.e.&nbsp;there’s an intercept and 1 slope. More explicitly:</p>
<p><span class="math display">\[
\mathbf{y} = \mathbf{X}\beta + \boldsymbol{\varepsilon},
\]</span></p>
<p><span class="math display">\[
\begin{bmatrix}
y_1 \\
y_2 \\
\vdots \\
y_n
\end{bmatrix}
=
\begin{bmatrix}
1 &amp; x_1 \\
1 &amp; x_2 \\
\vdots &amp; \vdots \\
1 &amp; x_n
\end{bmatrix}
\begin{bmatrix}
\beta_0 \\
\beta_1
\end{bmatrix}
+
\begin{bmatrix}
\varepsilon_1 \\
\varepsilon_2 \\
\vdots \\
\varepsilon_n
\end{bmatrix}.
\]</span></p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Definition: design matrix
</div>
</div>
<div class="callout-body-container callout-body">
<p><span class="math inline">\(\boldsymbol{X}\)</span> is called the “design matrix”, “covariate matrix”, “model matrix” or even sometimes the “data matrix”. It includes columns for each of the predictors and an intercept (if applicable).</p>
</div>
</div>
</section>
<section id="key-assumptions" class="level3">
<h3 class="anchored" data-anchor-id="key-assumptions">Key assumptions</h3>
<ol type="1">
<li>Linearity of <span class="math inline">\(Y\)</span> and <span class="math inline">\(X\)</span> (linearity applies to coefficients, <span class="math inline">\(\beta\)</span>).</li>
<li>Variation <span class="math inline">\(\sigma^2\)</span> in error term is independent of <span class="math inline">\(\boldsymbol{X}\)</span>.</li>
<li>Error terms are mutually independent</li>
<li>Error terms are normally distributed.</li>
</ol>
<p>These assumptions may be summarized:</p>
<p><span class="math display">\[
\boldsymbol{y}\sim N(\boldsymbol{X}\beta, \sigma^2 \boldsymbol{I})
\]</span></p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Definition: multiple regression
</div>
</div>
<div class="callout-body-container callout-body">
<p>If <span class="math inline">\(p &gt; 2\)</span>, we have more than one “predictor” variable and this is called <strong>multiple regression</strong>.</p>
</div>
</div>
</section>
<section id="picture-of-simple-vs-multiple-regression-offline" class="level3">
<h3 class="anchored" data-anchor-id="picture-of-simple-vs-multiple-regression-offline">Picture of simple vs multiple regression (offline)</h3>
</section>
</section>
<section id="linear-modeling-in-r" class="level2">
<h2 class="anchored" data-anchor-id="linear-modeling-in-r">Linear modeling in R</h2>
<p>Syntax:</p>
<ul>
<li><code>lm</code> stands for “linear model”</li>
<li>if variables <code>y</code> and <code>x</code> are objects in your environment, and you wish to fit the model <span class="math inline">\(y = \beta_0 + \beta_1 x_1\)</span>, you can run directly:</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lm</span>(y <span class="sc">~</span> x) <span class="co"># regresses  y on x</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>if <code>y</code> and <code>x</code> are columns in data frame <code>df</code>, then you must reference the data:</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lm</span>(y <span class="sc">~</span> x, <span class="at">data =</span> df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>to run multiple regression, <span class="math inline">\(y = \beta_0 + \beta_1 x_1 + \beta_2 x_2 + \beta_3 x_3 + \epsilon\)</span>, add in additional predictors with <code>+</code>:</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lm</span>(y <span class="sc">~</span> x1 <span class="sc">+</span> x2 <span class="sc">+</span> x3, <span class="at">data =</span> df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>to include an interaction effect, <span class="math inline">\(y = \beta_0 + \beta_1 x_1 + \beta_2 x_2 + \beta_3 x_1 x_2 + \epsilon\)</span></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lm</span>(y <span class="sc">~</span> x1 <span class="sc">*</span> x2, <span class="at">data =</span> df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>to drop the intercept term, e.g.&nbsp;the model is <span class="math inline">\(y = \beta_1 x_1 + \epsilon\)</span>, you have a couple options, each is equivalent:</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lm</span>(y <span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> x1)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">lm</span>(y <span class="sc">~</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">+</span> x1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>to look at the design matrix, you can run</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">model.matrix</span>(<span class="fu">lm</span>(y <span class="sc">~</span> x1 <span class="sc">+</span> x2))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Exercise
</div>
</div>
<div class="callout-body-container callout-body">
<p>Simulate some data, e.g.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">613</span>)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>N <span class="ot">&lt;-</span> <span class="dv">20</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(N)</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>x1 <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(N)</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>x2 <span class="ot">&lt;-</span> <span class="fu">runif</span>(N)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>and try out some of the functions above.</p>
</div>
</div>
<p>You can examine the output of your fitted model:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>model1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> x1 <span class="sc">+</span> x2)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model1)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>broom<span class="sc">::</span><span class="fu">glance</span>(model1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="fitted-model-output" class="level2">
<h2 class="anchored" data-anchor-id="fitted-model-output">Fitted model output</h2>
<ul>
<li>The <code>Estimate</code> output of <code>summary()</code> shows the least squares estimator <span class="math inline">\(\hat{\beta}\)</span> of <span class="math inline">\(\beta\)</span>.</li>
<li><code>Std Error</code> is SE(<span class="math inline">\(\hat{\beta}\)</span>)</li>
<li><code>t value</code> is the t-statistic for <span class="math inline">\(H_0: \beta = 0\)</span> vs alternative <span class="math inline">\(H_A: \beta \neq 0\)</span></li>
<li><code>Pr(&gt; |t|)</code> is the p-value for this test.</li>
<li>The function <code>glance</code> from the <code>broom</code> package, <code>broom::glance()</code> outputs additional summaries, including <code>sigma</code>, which shows the least squares estimator <span class="math inline">\(\hat{\sigma}^2\)</span> of <span class="math inline">\(\sigma^2\)</span></li>
</ul>
<p>Mathematically,</p>
<p><span class="math display">\[
\begin{aligned}
\hat{\beta} &amp;= (\boldsymbol{X}^T \boldsymbol{X})^{-1} \boldsymbol{X}^t \boldsymbol{y}\\
\hat{\sigma}^2 &amp;= MSE = \frac{\sum_{i=1}^n (y_i - \hat{y}_i)^2}{n-p}\\
\hat{y}_i &amp;= E[y_i | x_i, \hat{\beta}, \hat{\sigma}^2] = \hat{\beta}x_i ~~~(\text{predicted outcome})\\
e_i &amp;= y_i - \hat{y}_i ~~~(\text{residual})
\end{aligned}
\]</span></p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Remember: <span class="math inline">\(p\)</span> is the number of columns in the model matrix, including the intercept if present.</li>
<li>We say <span class="math inline">\(\hat{\sigma}^2\)</span> has <span class="math inline">\(n-p\)</span> degrees of freedom (df).</li>
</ul>
</div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Exercise
</div>
</div>
<div class="callout-body-container callout-body">
<p>Check the formulas above in R. Hint: <code>t(X)</code> is the transpose of a matrix “X”. <code>%*%</code> multiples matrices in R. <code>solve()</code> inverts them.</p>
<div class="cell">
<details class="code-fold">
<summary>Click to view solution</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>X <span class="ot">&lt;-</span>  <span class="fu">model.matrix</span>(<span class="fu">lm</span>(y <span class="sc">~</span> x1 <span class="sc">+</span> x2))</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>betaHat <span class="ot">&lt;-</span> <span class="fu">solve</span>(<span class="fu">t</span>(X) <span class="sc">%*%</span> X) <span class="sc">%*%</span> <span class="fu">t</span>(X) <span class="sc">%*%</span> y</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>betaHat</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                  [,1]
(Intercept)  0.6228850
x1           0.2531371
x2          -0.5065098</code></pre>
</div>
<details class="code-fold">
<summary>Click to view solution</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">nrow</span>(X)</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ncol</span>(X)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>yHat <span class="ot">&lt;-</span> X <span class="sc">%*%</span> betaHat </span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>s2 <span class="ot">&lt;-</span> <span class="fu">t</span>(y <span class="sc">-</span> yHat) <span class="sc">%*%</span> (y <span class="sc">-</span> yHat) <span class="sc">/</span> (n<span class="sc">-</span>p)</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a><span class="fu">sqrt</span>(s2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>         [,1]
[1,] 1.079028</code></pre>
</div>
</div>
</div>
</div>
</section>
<section id="return-to-metabolic-assay-data" class="level2">
<h2 class="anchored" data-anchor-id="return-to-metabolic-assay-data">Return to Metabolic Assay Data</h2>
<section id="linear-model-without-batch-pair-effects" class="level3">
<h3 class="anchored" data-anchor-id="linear-model-without-batch-pair-effects">Linear model without batch (pair) effects</h3>
<ul>
<li>genotype (g)</li>
<li>dose category (d): level of <span class="math inline">\(\Delta G_{ATP}\)</span></li>
<li>substrate (s) and</li>
<li>pair (p)</li>
</ul>
<p>Together, these uniquely identify samples.</p>
<p>Model:</p>
<p><span class="math display">\[
\begin{aligned}
y_{p,d,s,g} &amp;= \beta_{d,s} + \delta_{d,s} I_{geno='TG'} + \epsilon_{p,d,s,g}\\
\epsilon_{p,d,s,g} &amp;\sim N(0, \sigma^2)
\end{aligned}
\]</span> Components of <span class="math inline">\(\delta\)</span> are offsets measuring the differential effect of the transgenic (relative to the WT or non-Tg) genotype.</p>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-2-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-1" role="tab" aria-controls="tabset-2-1" aria-selected="true" href="">Model</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-2" role="tab" aria-controls="tabset-2-2" aria-selected="false" href="">Summary (short)</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-3" role="tab" aria-controls="tabset-2-3" aria-selected="false" href="">Summary (long)</a></li></ul>
<div class="tab-content">
<div id="tabset-2-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-2-1-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Linear regression model without batch effect</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>lm.out0<span class="ot">&lt;-</span><span class="fu">lm</span>(y <span class="sc">~</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">+</span> dose<span class="sc">:</span>substrate <span class="sc">+</span> isTg<span class="sc">:</span>dose<span class="sc">:</span>substrate,<span class="at">data=</span>o2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-2-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-2-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">summary</span>(lm.out0)<span class="sc">$</span>coefficients)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                         Estimate Std. Error  t value      Pr(&gt;|t|)
dose-13.65:substrateGM  12432.658   342.1096 36.34116 5.234813e-108
dose-13.95:substrateGM   9280.110   342.1096 27.12613  2.481786e-80
dose-14.19:substrateGM   6936.095   342.1096 20.27448  7.246422e-57
dose-14.36:substrateGM   5447.063   342.1096 15.92198  4.365917e-41
dose-14.49:substrateGM   4401.238   342.1096 12.86500  4.495322e-30
dose-13.65:substrateOcM  5210.492   374.7623 13.90346  8.962676e-34</code></pre>
</div>
</div>
</div>
<div id="tabset-2-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-3-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lm.out0)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = y ~ -1 + dose:substrate + isTg:dose:substrate, data = o2)

Residuals:
    Min      1Q  Median      3Q     Max 
-3230.0  -212.2   184.5   482.3  1375.0 

Coefficients:
                              Estimate Std. Error t value Pr(&gt;|t|)    
dose-13.65:substrateGM         12432.7      342.1  36.341  &lt; 2e-16 ***
dose-13.95:substrateGM          9280.1      342.1  27.126  &lt; 2e-16 ***
dose-14.19:substrateGM          6936.1      342.1  20.274  &lt; 2e-16 ***
dose-14.36:substrateGM          5447.1      342.1  15.922  &lt; 2e-16 ***
dose-14.49:substrateGM          4401.2      342.1  12.865  &lt; 2e-16 ***
dose-13.65:substrateOcM         5210.5      374.8  13.903  &lt; 2e-16 ***
dose-13.95:substrateOcM         4682.1      374.8  12.493  &lt; 2e-16 ***
dose-14.19:substrateOcM         4171.0      374.8  11.130  &lt; 2e-16 ***
dose-14.36:substrateOcM         3772.2      374.8  10.066  &lt; 2e-16 ***
dose-14.49:substrateOcM         3433.1      374.8   9.161  &lt; 2e-16 ***
dose-13.65:substratePcM         4862.0      374.8  12.974  &lt; 2e-16 ***
dose-13.95:substratePcM         4417.7      374.8  11.788  &lt; 2e-16 ***
dose-14.19:substratePcM         4001.1      374.8  10.676  &lt; 2e-16 ***
dose-14.36:substratePcM         3611.9      374.8   9.638  &lt; 2e-16 ***
dose-14.49:substratePcM         3286.5      374.8   8.770  &lt; 2e-16 ***
dose-13.65:substratePM         11111.3      342.1  32.479  &lt; 2e-16 ***
dose-13.95:substratePM          7145.1      342.1  20.886  &lt; 2e-16 ***
dose-14.19:substratePM          4936.5      342.1  14.429  &lt; 2e-16 ***
dose-14.36:substratePM          3735.7      342.1  10.920  &lt; 2e-16 ***
dose-14.49:substratePM          3013.6      342.1   8.809  &lt; 2e-16 ***
dose-13.65:substratePMOc       11757.1      342.1  34.366  &lt; 2e-16 ***
dose-13.95:substratePMOc        8579.5      342.1  25.078  &lt; 2e-16 ***
dose-14.19:substratePMOc        6416.1      342.1  18.755  &lt; 2e-16 ***
dose-14.36:substratePMOc        5067.2      342.1  14.812  &lt; 2e-16 ***
dose-14.49:substratePMOc        4157.4      342.1  12.152  &lt; 2e-16 ***
dose-13.65:substratePMPc        8581.5      342.1  25.084  &lt; 2e-16 ***
dose-13.95:substratePMPc        6712.7      342.1  19.621  &lt; 2e-16 ***
dose-14.19:substratePMPc        5301.0      342.1  15.495  &lt; 2e-16 ***
dose-14.36:substratePMPc        4348.3      342.1  12.710  &lt; 2e-16 ***
dose-14.49:substratePMPc        3634.0      342.1  10.622  &lt; 2e-16 ***
dose-13.65:substrateGM:isTg     1761.9      483.8   3.642 0.000323 ***
dose-13.95:substrateGM:isTg     1006.9      483.8   2.081 0.038337 *  
dose-14.19:substrateGM:isTg      779.2      483.8   1.610 0.108426    
dose-14.36:substrateGM:isTg      571.5      483.8   1.181 0.238481    
dose-14.49:substrateGM:isTg      444.9      483.8   0.920 0.358592    
dose-13.65:substrateOcM:isTg    1501.2      530.0   2.832 0.004955 ** 
dose-13.95:substrateOcM:isTg    1427.7      530.0   2.694 0.007491 ** 
dose-14.19:substrateOcM:isTg    1439.8      530.0   2.717 0.007006 ** 
dose-14.36:substrateOcM:isTg    1393.6      530.0   2.630 0.009023 ** 
dose-14.49:substrateOcM:isTg    1252.5      530.0   2.363 0.018802 *  
dose-13.65:substratePcM:isTg    2094.1      530.0   3.951 9.85e-05 ***
dose-13.95:substratePcM:isTg    1916.1      530.0   3.615 0.000356 ***
dose-14.19:substratePcM:isTg    1808.4      530.0   3.412 0.000740 ***
dose-14.36:substratePcM:isTg    1655.2      530.0   3.123 0.001978 ** 
dose-14.49:substratePcM:isTg    1432.6      530.0   2.703 0.007290 ** 
dose-13.65:substratePM:isTg     1023.8      483.8   2.116 0.035213 *  
dose-13.95:substratePM:isTg      277.2      483.8   0.573 0.567200    
dose-14.19:substratePM:isTg      319.4      483.8   0.660 0.509644    
dose-14.36:substratePM:isTg      328.1      483.8   0.678 0.498271    
dose-14.49:substratePM:isTg      326.3      483.8   0.674 0.500568    
dose-13.65:substratePMOc:isTg   2151.9      483.8   4.448 1.25e-05 ***
dose-13.95:substratePMOc:isTg   1265.3      483.8   2.615 0.009402 ** 
dose-14.19:substratePMOc:isTg    817.9      483.8   1.691 0.092038 .  
dose-14.36:substratePMOc:isTg    586.9      483.8   1.213 0.226093    
dose-14.49:substratePMOc:isTg    387.7      483.8   0.801 0.423673    
dose-13.65:substratePMPc:isTg   3432.9      483.8   7.095 1.06e-11 ***
dose-13.95:substratePMPc:isTg   2169.5      483.8   4.484 1.07e-05 ***
dose-14.19:substratePMPc:isTg   1364.8      483.8   2.821 0.005132 ** 
dose-14.36:substratePMPc:isTg    890.7      483.8   1.841 0.066668 .  
dose-14.49:substratePMPc:isTg    619.2      483.8   1.280 0.201632    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 838 on 280 degrees of freedom
Multiple R-squared:  0.9886,    Adjusted R-squared:  0.9862 
F-statistic: 406.2 on 60 and 280 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
</div>
</div>
</div>
</div>
<section id="diagnostic-plots-related-to-batch-checking-the-iid-errors-assumption" class="level4">
<h4 class="anchored" data-anchor-id="diagnostic-plots-related-to-batch-checking-the-iid-errors-assumption">Diagnostic plots related to batch (checking the IID errors assumption)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Graph residuals by batch for model 0</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>),<span class="at">las=</span><span class="dv">1</span>) <span class="co"># Plot the graphs in a 1 by 2 format</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(<span class="fu">residuals</span>(lm.out0)<span class="sc">~</span>o2<span class="sc">$</span>pair,<span class="at">main=</span><span class="st">"Resids Model 0 by Batch"</span>)</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="dv">0</span>,<span class="at">lty=</span><span class="dv">2</span>)</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(<span class="fu">residuals</span>(lm.out0)<span class="sc">~</span>o2<span class="sc">$</span>subject,<span class="at">main=</span><span class="st">"Resids Model 0 by Subject"</span>)</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="dv">0</span>,<span class="at">lty=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="multipleRegression_sta613_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="linear-model-with-batch-pair-effects" class="level3">
<h3 class="anchored" data-anchor-id="linear-model-with-batch-pair-effects">Linear model with batch (pair) effects</h3>
<p>Here, the model is:</p>
<p><span class="math display">\[
\begin{aligned}
y_{p,d,s,g} &amp;= \beta_{d,s} + \delta_{d,s} I_{geno='Tg'} + \gamma_p + \epsilon_{p,d,s,g}\\
\epsilon_{p,d,s,g} &amp;\sim N(0, \sigma^2)
\end{aligned}
\]</span></p>
<p>As above,</p>
<ul>
<li>the components of <span class="math inline">\(\delta\)</span> are offsets measuring the differential effect of the transgenic (relative to the non-Tg) genotype.</li>
<li><em>New</em>: components of <span class="math inline">\(\gamma\)</span> are the pair-specific batch effects.</li>
</ul>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-3-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-3-1" role="tab" aria-controls="tabset-3-1" aria-selected="true" href="">Model</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-3-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-3-2" role="tab" aria-controls="tabset-3-2" aria-selected="false" href="">Summary (long)</a></li></ul>
<div class="tab-content">
<div id="tabset-3-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-3-1-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>lm.out1<span class="ot">&lt;-</span><span class="fu">lm</span>(y <span class="sc">~</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">+</span> dose<span class="sc">:</span>substrate <span class="sc">+</span> isTg<span class="sc">:</span>dose<span class="sc">:</span>substrate <span class="sc">+</span> pair, <span class="at">data=</span>o2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-3-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-3-2-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lm.out1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = y ~ -1 + dose:substrate + isTg:dose:substrate + 
    pair, data = o2)

Residuals:
     Min       1Q   Median       3Q      Max 
-1710.89  -310.17    21.84   326.76  1485.15 

Coefficients: (1 not defined because of singularities)
                              Estimate Std. Error t value Pr(&gt;|t|)    
pair1                          3651.69     220.82  16.537  &lt; 2e-16 ***
pair2                          3939.47     220.82  17.840  &lt; 2e-16 ***
pair3                          4186.01     220.82  18.957  &lt; 2e-16 ***
pair4                          4115.51     220.82  18.637  &lt; 2e-16 ***
pair5                          3796.37     220.82  17.192  &lt; 2e-16 ***
pair6                          2114.89     224.85   9.406  &lt; 2e-16 ***
dose-13.65:substrateGM         8798.67     299.81  29.348  &lt; 2e-16 ***
dose-13.95:substrateGM         5646.12     299.81  18.833  &lt; 2e-16 ***
dose-14.19:substrateGM         3302.10     299.81  11.014  &lt; 2e-16 ***
dose-14.36:substrateGM         1813.07     299.81   6.047 4.79e-09 ***
dose-14.49:substrateGM          767.25     299.81   2.559 0.011028 *  
dose-13.65:substrateOcM        1272.68     314.80   4.043 6.86e-05 ***
dose-13.95:substrateOcM         744.26     314.80   2.364 0.018761 *  
dose-14.19:substrateOcM         233.22     314.80   0.741 0.459407    
dose-14.36:substrateOcM        -165.57     314.80  -0.526 0.599335    
dose-14.49:substrateOcM        -504.71     314.80  -1.603 0.110016    
dose-13.65:substratePcM         924.19     314.80   2.936 0.003608 ** 
dose-13.95:substratePcM         479.88     314.80   1.524 0.128551    
dose-14.19:substratePcM          63.32     314.80   0.201 0.840729    
dose-14.36:substratePcM        -325.94     314.80  -1.035 0.301391    
dose-14.49:substratePcM        -651.26     314.80  -2.069 0.039496 *  
dose-13.65:substratePM         7477.30     299.81  24.940  &lt; 2e-16 ***
dose-13.95:substratePM         3511.15     299.81  11.711  &lt; 2e-16 ***
dose-14.19:substratePM         1302.47     299.81   4.344 1.97e-05 ***
dose-14.36:substratePM          101.69     299.81   0.339 0.734728    
dose-14.49:substratePM         -620.40     299.81  -2.069 0.039448 *  
dose-13.65:substratePMOc       8123.09     299.81  27.095  &lt; 2e-16 ***
dose-13.95:substratePMOc       4945.53     299.81  16.496  &lt; 2e-16 ***
dose-14.19:substratePMOc       2782.14     299.81   9.280  &lt; 2e-16 ***
dose-14.36:substratePMOc       1433.24     299.81   4.781 2.85e-06 ***
dose-14.49:substratePMOc        523.44     299.81   1.746 0.081941 .  
dose-13.65:substratePMPc       4947.53     299.81  16.502  &lt; 2e-16 ***
dose-13.95:substratePMPc       3078.66     299.81  10.269  &lt; 2e-16 ***
dose-14.19:substratePMPc       1667.03     299.81   5.560 6.37e-08 ***
dose-14.36:substratePMPc        714.32     299.81   2.383 0.017870 *  
dose-14.49:substratePMPc            NA         NA      NA       NA    
dose-13.65:substrateGM:isTg    1761.87     299.81   5.877 1.21e-08 ***
dose-13.95:substrateGM:isTg    1006.85     299.81   3.358 0.000895 ***
dose-14.19:substrateGM:isTg     779.16     299.81   2.599 0.009857 ** 
dose-14.36:substrateGM:isTg     571.54     299.81   1.906 0.057646 .  
dose-14.49:substrateGM:isTg     444.90     299.81   1.484 0.138965    
dose-13.65:substrateOcM:isTg   1501.18     328.42   4.571 7.35e-06 ***
dose-13.95:substrateOcM:isTg   1427.70     328.42   4.347 1.94e-05 ***
dose-14.19:substrateOcM:isTg   1439.79     328.42   4.384 1.66e-05 ***
dose-14.36:substrateOcM:isTg   1393.64     328.42   4.243 3.01e-05 ***
dose-14.49:substrateOcM:isTg   1252.48     328.42   3.814 0.000169 ***
dose-13.65:substratePcM:isTg   2094.15     328.42   6.376 7.63e-10 ***
dose-13.95:substratePcM:isTg   1916.05     328.42   5.834 1.52e-08 ***
dose-14.19:substratePcM:isTg   1808.41     328.42   5.506 8.40e-08 ***
dose-14.36:substratePcM:isTg   1655.19     328.42   5.040 8.46e-07 ***
dose-14.49:substratePcM:isTg   1432.62     328.42   4.362 1.82e-05 ***
dose-13.65:substratePM:isTg    1023.84     299.81   3.415 0.000734 ***
dose-13.95:substratePM:isTg     277.16     299.81   0.924 0.356057    
dose-14.19:substratePM:isTg     319.43     299.81   1.065 0.287599    
dose-14.36:substratePM:isTg     328.07     299.81   1.094 0.274787    
dose-14.49:substratePM:isTg     326.32     299.81   1.088 0.277355    
dose-13.65:substratePMOc:isTg  2151.92     299.81   7.178 6.62e-12 ***
dose-13.95:substratePMOc:isTg  1265.26     299.81   4.220 3.32e-05 ***
dose-14.19:substratePMOc:isTg   817.91     299.81   2.728 0.006780 ** 
dose-14.36:substratePMOc:isTg   586.94     299.81   1.958 0.051271 .  
dose-14.49:substratePMOc:isTg   387.65     299.81   1.293 0.197093    
dose-13.65:substratePMPc:isTg  3432.90     299.81  11.450  &lt; 2e-16 ***
dose-13.95:substratePMPc:isTg  2169.53     299.81   7.236 4.60e-12 ***
dose-14.19:substratePMPc:isTg  1364.77     299.81   4.552 7.98e-06 ***
dose-14.36:substratePMPc:isTg   890.74     299.81   2.971 0.003230 ** 
dose-14.49:substratePMPc:isTg   619.25     299.81   2.065 0.039813 *  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 519.3 on 275 degrees of freedom
Multiple R-squared:  0.9957,    Adjusted R-squared:  0.9947 
F-statistic: 983.5 on 65 and 275 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
</div>
</div>
</div>
</div>
<section id="diagnostic-plots-model-with-batch-adjustment" class="level4">
<h4 class="anchored" data-anchor-id="diagnostic-plots-model-with-batch-adjustment">Diagnostic plots: model with batch adjustment</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>),<span class="at">las=</span><span class="dv">1</span>) <span class="co"># Plot the graphs in a 1 by 2 format</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(<span class="fu">residuals</span>(lm.out1)<span class="sc">~</span>o2<span class="sc">$</span>pair,<span class="at">main=</span><span class="st">"Resids Model 1 by Batch"</span>)</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="dv">0</span>,<span class="at">lty=</span><span class="dv">2</span>)</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(<span class="fu">residuals</span>(lm.out1)<span class="sc">~</span>o2<span class="sc">$</span>subject,<span class="at">main=</span><span class="st">"Resids Model 1 by Subject"</span>)</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="dv">0</span>,<span class="at">lty=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="multipleRegression_sta613_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="other-residual-diagnostic-plots" class="level4">
<h4 class="anchored" data-anchor-id="other-residual-diagnostic-plots">Other residual diagnostic plots</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>),<span class="at">las=</span><span class="dv">1</span>)</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>r<span class="ot">&lt;-</span><span class="fu">residuals</span>(lm.out1)</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(r<span class="sc">~</span>o2<span class="sc">$</span>substrate,<span class="at">main=</span><span class="st">"Resids Model 1 by Substrate"</span>)</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="dv">0</span>,<span class="at">lty=</span><span class="dv">2</span>)</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(r<span class="sc">~</span>o2<span class="sc">$</span>geno,<span class="at">main=</span><span class="st">"Resids Model 1 by Genotype"</span>)</span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="dv">0</span>,<span class="at">lty=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="multipleRegression_sta613_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>),<span class="at">las=</span><span class="dv">1</span>)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(r<span class="sc">~</span>o2<span class="sc">$</span>dose,<span class="at">main=</span><span class="st">"Resids Model 1 by Dose"</span>)</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="dv">0</span>,<span class="at">lty=</span><span class="dv">2</span>)</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a><span class="fu">qqnorm</span>(r,<span class="at">main=</span><span class="st">"Model 1 Residuals"</span>)</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">a=</span><span class="dv">0</span>,<span class="at">b=</span><span class="fu">sd</span>(r),<span class="at">lty=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="multipleRegression_sta613_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="testing" class="level2">
<h2 class="anchored" data-anchor-id="testing">Testing</h2>
<section id="key-quantities" class="level3">
<h3 class="anchored" data-anchor-id="key-quantities">Key quantities:</h3>
<ul>
<li>Total sum of squares (TSS): <span class="math inline">\(\sum_{i=1}^n (y_i - \bar{y})^2\)</span></li>
<li>Residual sum of squares (RSS): <span class="math inline">\(\sum_{i=1}^n (y_i - \hat{y}_i)^2\)</span></li>
<li>Regression sum of squares (<span class="math inline">\(SS_{reg}\)</span>): TSS - RSS; TSS = <span class="math inline">\(SS_{reg}\)</span> + RSS</li>
<li>Mean squared error (MSE): <span class="math inline">\(\frac{RSS}{n - p}\)</span></li>
<li>Here <span class="math inline">\(p\)</span> is the number of columns in the design matrix including the column of 1s for the intercept (if present).</li>
</ul>
</section>
<section id="f-test-for-the-comparison-of-nested-models" class="level3">
<h3 class="anchored" data-anchor-id="f-test-for-the-comparison-of-nested-models">F-test for the comparison of nested models</h3>
<ul>
<li><p>Test for the regression</p>
<ul>
<li>“Null” model includes only an intercept</li>
<li>“Full” model adds to this <span class="math inline">\(p-1\)</span> covariates:</li>
</ul>
<p><span class="math display">\[
F = \frac{SS_{reg}/p}{MSE}
\]</span> This has an F-distribution with <span class="math inline">\((p, n-p)\)</span> degrees of freedom.</p></li>
<li><p>For pair of nested models</p>
<ul>
<li>“Null” model includes a subset of the covariates included in the “full” model:</li>
</ul>
<p><span class="math display">\[
F = \frac{(SS_{reg,F} - SS_{reg,N}) / q}{MSE_{F}}
\]</span></p>
<ul>
<li>This has an F-distribution with <span class="math inline">\((q, n-p)\)</span> degrees of freedom, where <span class="math inline">\(q\)</span> is the difference in the number of predictors between the two models.</li>
</ul></li>
</ul>
</section>
<section id="example-f-test-for-the-genotype-effect" class="level3">
<h3 class="anchored" data-anchor-id="example-f-test-for-the-genotype-effect">Example: F-test for the genotype effect</h3>
<ul>
<li>Null model:</li>
</ul>
<p><span class="math display">\[
\begin{aligned}
y_{p,d,s,g} &amp;= \beta_{d,s} + \gamma_p + \epsilon_{p, d, g, s}\\
\epsilon_{p,d,s,g} &amp;\sim N(0, \sigma^2)
\end{aligned}
\]</span></p>
<ul>
<li>“Alternate” model:</li>
</ul>
<p><span class="math display">\[
\begin{aligned}
y_{p,d,s,g} &amp;= \beta_{d,s} + \delta_{d,s} I_{geno='Tg'} + \gamma_p + \epsilon_{p, d, g, s}\\
\epsilon_{p,d,s,g} &amp;\sim N(0, \sigma^2)
\end{aligned}
\]</span></p>
<p>In the null model the offsets measuring the differential effect of the transgenic genotype are set to zero.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>lm.null<span class="ot">&lt;-</span><span class="fu">lm</span>(y <span class="sc">~</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">+</span> dose<span class="sc">:</span>substrate <span class="sc">+</span> pair,<span class="at">data=</span>o2)</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(lm.null,lm.out1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Analysis of Variance Table

Model 1: y ~ -1 + dose:substrate + pair
Model 2: y ~ -1 + dose:substrate + isTg:dose:substrate + pair
  Res.Df       RSS Df Sum of Sq      F    Pr(&gt;F)    
1    305 239993485                                  
2    275  74153877 30 165839608 20.501 &lt; 2.2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<p>Large F-statistics yields a small p-value, which implies evidence against the null. Genotype appears to be an important predictor.</p>
</section>
</section>
<section id="dose-as-continuous-covariate" class="level2">
<h2 class="anchored" data-anchor-id="dose-as-continuous-covariate">Dose as continuous covariate</h2>
<p>Model for continuous dose:</p>
<p><span class="math display">\[
\begin{aligned}
y_{p,d,s,g} &amp;= \alpha_s + \beta_{s}d + \delta_sI_{geno = 'Tg'} + \mu_s I_{geno='Tg'}d + \gamma_p + \epsilon_{p, d, g, s}\\
\epsilon_{p,d,s,g} &amp;\sim N(0, \sigma^2)
\end{aligned}
\]</span></p>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-4-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-4-1" role="tab" aria-controls="tabset-4-1" aria-selected="true" href="">Model</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-4-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-4-2" role="tab" aria-controls="tabset-4-2" aria-selected="false" href="">Summary</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-4-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-4-3" role="tab" aria-controls="tabset-4-3" aria-selected="false" href="">Diagnostics</a></li></ul>
<div class="tab-content">
<div id="tabset-4-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-4-1-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>o2<span class="sc">$</span>doseX<span class="ot">&lt;-</span><span class="fu">as.numeric</span>(<span class="fu">as.character</span>(o2<span class="sc">$</span>dose))</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Linear regression model with batch effect</span></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>lm.out3<span class="ot">&lt;-</span><span class="fu">lm</span>(y <span class="sc">~</span> </span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>              <span class="sc">-</span><span class="dv">1</span> <span class="sc">+</span> substrate <span class="sc">+</span> doseX<span class="sc">:</span>substrate <span class="sc">+</span></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>              isTg<span class="sc">:</span>substrate <span class="sc">+</span> isTg<span class="sc">:</span>doseX<span class="sc">:</span>substrate <span class="sc">+</span> pair,<span class="at">data=</span>o2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-4-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-4-2-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lm.out3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = y ~ -1 + substrate + doseX:substrate + isTg:substrate + 
    isTg:doseX:substrate + pair, data = o2)

Residuals:
     Min       1Q   Median       3Q      Max 
-1735.04  -367.76    14.55   361.94  1756.76 

Coefficients:
                         Estimate Std. Error t value Pr(&gt;|t|)    
substrateGM              143296.1     4891.8  29.293  &lt; 2e-16 ***
substrateOcM              33788.5     5358.6   6.305 9.84e-10 ***
substratePcM              30018.8     5358.6   5.602 4.67e-08 ***
substratePM              141953.7     4891.8  29.019  &lt; 2e-16 ***
substratePMOc            135269.5     4891.8  27.652  &lt; 2e-16 ***
substratePMPc             88945.2     4891.8  18.182  &lt; 2e-16 ***
pair2                       287.8      103.7   2.775  0.00586 ** 
pair3                       534.3      103.7   5.152 4.59e-07 ***
pair4                       463.8      103.7   4.472 1.09e-05 ***
pair5                       144.7      103.7   1.395  0.16401    
pair6                     -1536.8      118.3 -12.996  &lt; 2e-16 ***
substrateGM:doseX          9596.5      346.1  27.724  &lt; 2e-16 ***
substrateOcM:doseX         2110.8      379.2   5.567 5.61e-08 ***
substratePcM:doseX         1859.4      379.2   4.904 1.52e-06 ***
substratePM:doseX          9622.6      346.1  27.800  &lt; 2e-16 ***
substratePMOc:doseX        9064.0      346.1  26.186  &lt; 2e-16 ***
substratePMPc:doseX        5889.9      346.1  17.016  &lt; 2e-16 ***
substrateGM:isTg          22297.3     6917.4   3.223  0.00140 ** 
substrateOcM:isTg          4729.9     7577.7   0.624  0.53296    
substratePcM:isTg         12083.5     7577.7   1.595  0.11181    
substratePM:isTg          10674.0     6917.4   1.543  0.12383    
substratePMOc:isTg        30206.4     6917.4   4.367 1.72e-05 ***
substratePMPc:isTg        49273.5     6917.4   7.123 7.37e-12 ***
substrateGM:doseX:isTg     1513.6      489.5   3.092  0.00217 ** 
substrateOcM:doseX:isTg     235.5      536.2   0.439  0.66086    
substratePcM:doseX:isTg     729.2      536.2   1.360  0.17486    
substratePM:doseX:isTg      723.3      489.5   1.478  0.14052    
substratePMOc:doseX:isTg   2064.3      489.5   4.217 3.25e-05 ***
substratePMPc:doseX:isTg   3367.6      489.5   6.880 3.31e-11 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 568.1 on 311 degrees of freedom
Multiple R-squared:  0.9942,    Adjusted R-squared:  0.9937 
F-statistic:  1839 on 29 and 311 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
</div>
</div>
<div id="tabset-4-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-4-3-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="at">las =</span> <span class="dv">1</span>)</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>r <span class="ot">&lt;-</span> <span class="fu">residuals</span>(lm.out3)</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(r <span class="sc">~</span> o2<span class="sc">$</span>substrate, <span class="at">main =</span> <span class="st">"Resids Model 3 by Substrate"</span>)</span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">0</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(r <span class="sc">~</span> o2<span class="sc">$</span>geno, <span class="at">main =</span> <span class="st">"Resids Model 3 by Genotype"</span>)</span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">0</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="multipleRegression_sta613_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="at">las =</span> <span class="dv">1</span>)</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(r <span class="sc">~</span> o2<span class="sc">$</span>dose, <span class="at">main =</span> <span class="st">"Resids Model 3 by Dose"</span>)</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">0</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a><span class="fu">qqnorm</span>(r, <span class="at">main =</span> <span class="st">"Model 3 Residuals"</span>)</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">a =</span> <span class="dv">0</span>, <span class="at">b =</span> <span class="fu">sd</span>(r), <span class="at">lty =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="multipleRegression_sta613_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
</div>
</div>
<section id="curvilinear-dose" class="level3">
<h3 class="anchored" data-anchor-id="curvilinear-dose">Curvilinear dose</h3>
<p><span class="math display">\[
\begin{aligned}
y_{p,d,s,g} &amp;= \alpha_s + \beta_{s}\text{poly}(d,2) + \delta_sI_{geno = 'Tg'} + \mu_s I_{geno='Tg'}\text{poly}(d,2) + \gamma_p + \epsilon_{p, d, g, s}\\
\epsilon_{p,d,s,g} &amp;\sim N(0, \sigma^2)
\end{aligned}
\]</span></p>
<ul>
<li>What is <code>poly(d, 2)</code>?</li>
</ul>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-5-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-5-1" role="tab" aria-controls="tabset-5-1" aria-selected="true" href="">Model</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-5-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-5-2" role="tab" aria-controls="tabset-5-2" aria-selected="false" href="">Summary</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-5-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-5-3" role="tab" aria-controls="tabset-5-3" aria-selected="false" href="">Diagnostics</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-5-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-5-4" role="tab" aria-controls="tabset-5-4" aria-selected="false" href="">F-test</a></li></ul>
<div class="tab-content">
<div id="tabset-5-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-5-1-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>lm.out4<span class="ot">&lt;-</span><span class="fu">lm</span>(y <span class="sc">~</span> </span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>              <span class="sc">-</span><span class="dv">1</span> <span class="sc">+</span> substrate <span class="sc">+</span> <span class="fu">poly</span>(doseX,<span class="dv">2</span>)<span class="sc">:</span>substrate <span class="sc">+</span> </span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>              isTg<span class="sc">:</span>substrate <span class="sc">+</span> isTg<span class="sc">:</span><span class="fu">poly</span>(doseX,<span class="dv">2</span>)<span class="sc">:</span>substrate <span class="sc">+</span> pair,<span class="at">data=</span>o2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-5-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-5-2-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lm.out4)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = y ~ -1 + substrate + poly(doseX, 2):substrate + 
    isTg:substrate + isTg:poly(doseX, 2):substrate + pair, data = o2)

Residuals:
     Min       1Q   Median       3Q      Max 
-1717.94  -308.34    22.71   330.26  1499.14 

Coefficients:
                                   Estimate Std. Error t value Pr(&gt;|t|)    
substrateGM                         7717.13     108.95  70.830  &lt; 2e-16 ***
substrateOcM                        3967.66     115.45  34.367  &lt; 2e-16 ***
substratePcM                        3749.73     115.45  32.480  &lt; 2e-16 ***
substratePM                         6006.13     108.95  55.126  &lt; 2e-16 ***
substratePMOc                       7213.17     108.95  66.205  &lt; 2e-16 ***
substratePMPc                       5733.20     108.95  52.621  &lt; 2e-16 ***
pair2                                287.78      91.27   3.153  0.00178 ** 
pair3                                534.33      91.27   5.854 1.26e-08 ***
pair4                                463.82      91.27   5.082 6.60e-07 ***
pair5                                144.68      91.27   1.585  0.11398    
pair6                              -1536.80     104.06 -14.768  &lt; 2e-16 ***
substrateGM:poly(doseX, 2)1        53018.97    1682.94  31.504  &lt; 2e-16 ***
substrateOcM:poly(doseX, 2)1       11661.62    1843.57   6.326 9.19e-10 ***
substratePcM:poly(doseX, 2)1       10272.69    1843.57   5.572 5.62e-08 ***
substratePM:poly(doseX, 2)1        53163.12    1682.94  31.589  &lt; 2e-16 ***
substratePMOc:poly(doseX, 2)1      50077.19    1682.94  29.756  &lt; 2e-16 ***
substratePMPc:poly(doseX, 2)1      32540.54    1682.94  19.336  &lt; 2e-16 ***
substrateGM:poly(doseX, 2)2         2557.53    1682.94   1.520  0.12965    
substrateOcM:poly(doseX, 2)2        -867.09    1843.57  -0.470  0.63846    
substratePcM:poly(doseX, 2)2       -1116.77    1843.57  -0.606  0.54513    
substratePM:poly(doseX, 2)2         8422.45    1682.94   5.005 9.57e-07 ***
substratePMOc:poly(doseX, 2)2       3796.90    1682.94   2.256  0.02479 *  
substratePMPc:poly(doseX, 2)2        831.22    1682.94   0.494  0.62173    
substrateGM:isTg                     912.86     129.08   7.072 1.08e-11 ***
substrateOcM:isTg                   1402.96     141.40   9.922  &lt; 2e-16 ***
substratePcM:isTg                   1781.28     141.40  12.598  &lt; 2e-16 ***
substratePM:isTg                     454.96     129.08   3.525  0.00049 ***
substratePMOc:isTg                  1041.94     129.08   8.072 1.70e-14 ***
substratePMPc:isTg                  1695.44     129.08  13.135  &lt; 2e-16 ***
substrateGM:poly(doseX, 2)1:isTg    8362.52    2380.04   3.514  0.00051 ***
substrateOcM:poly(doseX, 2)1:isTg   1301.03    2607.20   0.499  0.61814    
substratePcM:poly(doseX, 2)1:isTg   4028.73    2607.20   1.545  0.12335    
substratePM:poly(doseX, 2)1:isTg    3996.22    2380.04   1.679  0.09419 .  
substratePMOc:poly(doseX, 2)1:isTg 11404.92    2380.04   4.792 2.61e-06 ***
substratePMPc:poly(doseX, 2)1:isTg 18605.69    2380.04   7.817 9.28e-14 ***
substrateGM:poly(doseX, 2)2:isTg    1726.31    2380.04   0.725  0.46882    
substrateOcM:poly(doseX, 2)2:isTg   -494.59    2607.20  -0.190  0.84967    
substratePcM:poly(doseX, 2)2:isTg   -838.50    2607.20  -0.322  0.74797    
substratePM:poly(doseX, 2)2:isTg    3078.41    2380.04   1.293  0.19686    
substratePMOc:poly(doseX, 2)2:isTg  1864.95    2380.04   0.784  0.43391    
substratePMPc:poly(doseX, 2)2:isTg  2145.02    2380.04   0.901  0.36818    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 499.9 on 299 degrees of freedom
Multiple R-squared:  0.9957,    Adjusted R-squared:  0.9951 
F-statistic:  1682 on 41 and 299 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
</div>
</div>
<div id="tabset-5-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-5-3-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>),<span class="at">las=</span><span class="dv">1</span>)</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>r<span class="ot">&lt;-</span><span class="fu">residuals</span>(lm.out4)</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(r<span class="sc">~</span>o2<span class="sc">$</span>substrate,<span class="at">main=</span><span class="st">"Resids Model 4 by Substrate"</span>)</span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="dv">0</span>,<span class="at">lty=</span><span class="dv">2</span>)</span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(r<span class="sc">~</span>o2<span class="sc">$</span>geno,<span class="at">main=</span><span class="st">"Resids Model 4 by Genotype"</span>)</span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="dv">0</span>,<span class="at">lty=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="multipleRegression_sta613_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>),<span class="at">las=</span><span class="dv">1</span>)</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(r<span class="sc">~</span>o2<span class="sc">$</span>dose,<span class="at">main=</span><span class="st">"Resids Model 4 by Dose"</span>)</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="dv">0</span>,<span class="at">lty=</span><span class="dv">2</span>)</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a><span class="fu">qqnorm</span>(r,<span class="at">main=</span><span class="st">"Model 4 Residuals"</span>)</span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">a=</span><span class="dv">0</span>,<span class="at">b=</span><span class="fu">sd</span>(r),<span class="at">lty=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="multipleRegression_sta613_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-5-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-5-4-tab">
<ul>
<li>Test for added value of quadratic term</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(lm.out3,lm.out4)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Analysis of Variance Table

Model 1: y ~ -1 + substrate + doseX:substrate + isTg:substrate + isTg:doseX:substrate + 
    pair
Model 2: y ~ -1 + substrate + poly(doseX, 2):substrate + isTg:substrate + 
    isTg:poly(doseX, 2):substrate + pair
  Res.Df       RSS Df Sum of Sq      F    Pr(&gt;F)    
1    311 100356456                                  
2    299  74722465 12  25633991 8.5478 5.843e-14 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
</div>
</div>
</div>
<p>Other questions / issues</p>
<ul>
<li>Are the dose by substrate interaction terms needed?</li>
<li>What about subject effects? Subject and genotype are confounded.</li>
<li>Anything else?</li>
</ul>
</section>
</section>
<section id="references" class="level2 unnumbered">


</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-fisher2018mitochondrial" class="csl-entry" role="listitem">
Fisher-Wellman, Kelsey H, Michael T Davidson, Tara M Narowski, Chien-Te Lin, Timothy R Koves, and Deborah M Muoio. 2018. <span>“Mitochondrial Diagnostics: A Multiplexed Assay Platform for Comprehensive Assessment of Mitochondrial Energy Fluxes.”</span> <em>Cell Reports</em> 24 (13): 3593–3606. <a href="https://pubmed.ncbi.nlm.nih.gov/30257218/">https://pubmed.ncbi.nlm.nih.gov/30257218/</a>.
</div>
</div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>